servicedef/sdk_config.go

```diff
@@ -104,3 +105,55 @@ type SDKConfigWrapper struct {
 	Name    string `json:"name"`
 	Version string `json:"version"`
 }
+
+type SDKConfigDataSystem struct {
+	Synchronizers []SDKConfigDataSystemDataSource `json:"synchronizers,omitempty"`
+	Initializers  []SDKConfigDataSystemDataSource `json:"initializers,omitempty"`
+	Persistence   SDKConfigDataSystemPersistence  `json:"persistence,omitempty"`
+}
+
+type SDKConfigDataSystemDataSource struct {
+	Type string `json:"type"`
+	DataSystemDataSourceConfig
+}
+
+type DataSystemDataSourceConfig interface{}
+
+type DataSystemStreamingDataSourceConfig struct {
+	SDKConfigStreamingParams
+}
+
+type DataSystemPollingDataSourceConfig struct {
+	SDKConfigPollingParams
+}
+
+type SDKConfigDataSystemPersistence struct {
+	Store SDKConfigDataSystemPersistenceStore `json:"store"`
+	Cache SDKConfigDataSystemPersistenceCache `json:"cache"`
+}
+
+type SDKConfigDataSystemPersistenceType string
+
+const (
+	Redis    SDKConfigDataSystemPersistenceType = "redis"
```

Comment:
Another style of doing this is `Redis = SDKConfigDataSystemPersistenceType("redis")`

I think they are equivalent.

Reply:
Updated.